# HG changeset patch
# User Maxim Dounin <mdounin@mdounin.ru>
# Date 1313148201 -14400
# Node ID 6234b54267ea13057974373d5f3e461795bcd431
# Parent  7ea1bba9a4f6fedf1df77201fa15abc0ef07639d
Configure: catch up with new Linux version numbering.

Catch up with new Linux version numbering scheme as announced at [1] and
supress unrecognized versions to actually use default 0.

[1] https://lkml.org/lkml/2011/5/29/204

diff --git a/auto/os/linux b/auto/os/linux
--- a/auto/os/linux
+++ b/auto/os/linux
@@ -18,7 +18,9 @@ CC_AUX_FLAGS="$cc_aux_flags -D_GNU_SOURC
 # Linux kernel version
 
 version=$((`uname -r \
-       | sed 's/^\([0-9]*\)\.\([0-9]*\)\.\([0-9]*\).*/\1*256*256+\2*256+\3/'`))
+    | sed -n -e 's/^\([0-9][0-9]*\)\.\([0-9][0-9]*\)\.\([0-9][0-9]*\).*/ \
+                                                 \1*256*256+\2*256+\3/p' \
+             -e 's/^\([0-9][0-9]*\)\.\([0-9][0-9]*\).*/\1*256*256+\2*256/p'`))
 
 version=${version:-0}
 
